language-plutus-core/src/Language/PlutusCore/Constant/Typed.hs

```diff
@@ -131,11 +138,32 @@ data TypedBuiltinValue size a = TypedBuiltinValue (TypedBuiltin size a) a
 data TypeScheme size a r where
     TypeSchemeBuiltin :: TypedBuiltin size a -> TypeScheme size a a
     TypeSchemeArrow   :: TypeScheme size a q -> TypeScheme size b r -> TypeScheme size (a -> b) r
+    TypeSchemeAllType
+        :: (KnownSymbol text, KnownNat uniq)
```

Comment:
I get it, I think, but this is still a bit scary :fearful:

Reply:
The new constructor, a bit surprisingly, does not introduce any complication at all. You basically never need to think how to instantiate the continuation, because it always gets instantiated by the same `TypedBuiltinDyn` (just do `Ctrl-f `TypeSchemeAllType` on the changes tab and see yourself). So what worked previously, still works, and what didn't work (`error "Not implemented."`) still does not work.

It won't always be like that, though, because right now we manually provide type uniques, which is silly, and once we stop doing that, there will be corner cases, for example you won't be able to always compute the resulting type of a `TypeScheme`, because as a standalone type it may contain a free variable (we need this only for tests and do not really care much), but I do expect the final design to be nice to deal with.